pennylane/transforms/commutation_dag.py

```diff
@@ -0,0 +1,1049 @@
+        all_zeros = np.allclose(np.mod(operation1.data, 2 * np.pi), 0)
+        if all_zeros:
+            if operation2.name not in ["Barrier", "WireCut"]:
+                return True
+            return False
+    return None
+
+
+def two_non_simplified_rotations(operation1, operation2):
+    r"""Check commutation for two rotations that were not simplified.
+
+    Args:
+        operation1 (pennylane.Operation): First operation.
+        operation2 (pennylane.Operation): Second operation.
+
+    Returns:
+         Bool: True if commutation, False otherwise.
+    """
+    # Two non simplified rotations
+    if (operation1.name in ["U2", "U3", "Rot", "CRot"]) and (
+        operation2.name in ["U2", "U3", "Rot", "CRot"]
+    ):
+        target_wires_1 = qml.wires.Wires(
+            [w for w in operation1.wires if w not in operation1.control_wires]
+        )
+        target_wires_2 = qml.wires.Wires(
+            [w for w in operation2.wires if w not in operation2.control_wires]
+        )
+
+        if operation1.name == "CRot":
+            if not intersection(target_wires_1, operation2.wires):
+                return bool(commutation_map["ctrl"][position[operation2.name]])
+            return np.all(
+                np.allclose(
+                    np.matmul(
+                        qml.Rot(*operation1.data, wires=target_wires_1).get_matrix(),
+                        operation2.get_matrix(),
+                    ),
+                    np.matmul(
+                        operation2.get_matrix(),
+                        qml.Rot(*operation1.data, wires=target_wires_1).get_matrix(),
+                    ),
+                )
+            )
+
+        if operation2.name == "CRot":
+            if not intersection(target_wires_2, operation1.wires):
+                return bool(commutation_map[operation1.name][position["ctrl"]])
+            return np.all(
+                np.allclose(
+                    np.matmul(
+                        qml.Rot(*operation2.data, wires=target_wires_2).get_matrix(),
+                        operation1.get_matrix(),
+                    ),
+                    np.matmul(
+                        operation1.get_matrix(),
+                        qml.Rot(*operation2.data, wires=target_wires_2).get_matrix(),
+                    ),
+                )
+            )
+
+        return np.all(
+            np.allclose(
+                np.matmul(
+                    operation1.get_matrix(),
+                    operation2.get_matrix(),
+                ),
+                np.matmul(
+                    operation2.get_matrix(),
+                    operation1.get_matrix(),
+                ),
+            )
+        )
+    return None
+
+
+not_supported_operations = [
+    "PauliRot",
+    "QubitDensityMatrix",
+    "CVNeuralNetLayers",
+    "ApproxTimeEvolution",
+    "ArbitraryUnitary" "CommutingEvolution",
+    "DisplacementEmbedding",
+    "SqueezingEmbedding",
+]
+non_commuting_operations = [
```

Comment:
As in, these will never commute with anything?

Reply:
Some might, e.g., if they are the identity (parameter=0)? 🙂

Reply:
I think they might have cases where it commutes but not many. My reasoning is that most of them are not supported on any devices and will be decomposed. I guess I will let it as it is now, as it would require much work to check every different possibility and the code will become even more unreadable.

Reply:
got it!